pkgs/development/python-modules/xgrammar/default.nix

```diff
@@ -55,16 +55,15 @@ buildPythonPackage rec {
     torch
     transformers
   ]
-  ++ lib.optionals (stdenv.hostPlatform.isLinux && stdenv.hostPlatform.isx86_64) [
-    triton
-  ];
+  ++ lib.optional (lib.meta.availableOn stdenv.hostPlatform triton) triton
+  ++ lib.optional (lib.meta.availableOn stdenv.hostPlatform mlx-lm) mlx-lm;
 
   nativeCheckInputs = [
     pytestCheckHook
     writableTmpDirAsHomeHook
   ];
 
-  NIX_CFLAGS_COMPILE = toString [
+  NIX_CFLAGS_COMPILE = lib.optionals stdenv.cc.isGNU [
```

Comment:
This will make `NIX_CFLAGS_COMPILE` not a string on Linux. Otherwise LGTM.

Reply:
Will fix that